### ISSUE TITLE
Please investigate prebuilding stdlib with CGO flags enabled.

### ISSUE THREAD

**Issue by sodre** [conda-forge/go-feedstock#14]
Would it be possible to "precompile" stdlib with CGO_FLAGS enabled? 

I ran into an issue while compiling the kubernetes package from source where I had to "chmod 555" the $GOROOT directory or it would end up packaging the recompiled version of stdlib. The drawback of the changed chmod is the following warning and longer compile time...

```
  kube::log::status "+++ Warning: stdlib pkg with cgo flag not found.";
  kube::log::status "+++ Warning: stdlib pkg cannot be rebuilt since ${go_root_dir}/pkg is not writable by `whoami`";
  kube::log::status "+++ Warning: Make ${go_root_dir}/pkg writable for `whoami` for a one-time stdlib install, Or"
  kube::log::status "+++ Warning: Rebuild stdlib using the command 'CGO_ENABLED=0 go install -a -installsuffix cgo std'";
  kube::log::status "+++ Falling back to go build, which is slower";
```

I imagine the `go install` line would come in just before the end of the build.sh recipe.

**Comment by nehaljwani** [comment]
@sodre Could you send a PR for this?

**Comment by sodre** [comment]
@nehaljwani, no problem. I created the PRs in #16 #15 and #17.

**Comment by nehaljwani** [comment]
@sodre I have merged your patches, thank you for the PRs.

Could you please explain why the value of CGO_ENABLED was set to 0?

**Comment by sodre** [comment]
@nehaljwani, I don't have a direct answer other than that is the line that the kubernetes build suggested running to create the stdlib install.

```
  kube::log::status "+++ Warning: Rebuild stdlib using the command 'CGO_ENABLED=0 go install -a -installsuffix cgo std'";
```

**Comment by sodre** [comment]
This was addressed in #5474

**Comment by jakirkham** [comment]
Is it worth adding to a FAQ in the docs?
